pennylane_qulacs/qulacs_device.py

```diff
@@ -53,167 +93,215 @@ def hermitian(*args):
     A = np.asarray(args[0])
 
     if A.shape[0] != A.shape[1]:
-        raise ValueError('Expectation must be a square matrix.')
+        raise ValueError("Expectation must be a square matrix.")
 
     if not np.allclose(A, A.conj().T):
-        raise ValueError('Expectation must be Hermitian.')
+        raise ValueError("Expectation must be Hermitian.")
 
     return A
 
 
-class QulacsDevice(Device):
+# tolerance for numerical errors
+tolerance = 1e-10
+
+
+class QulacsDevice(QubitDevice):
     """Qulacs device"""
-    name = 'Qulacs device'
-    short_name = 'qulacs.simulator'
-    pennylane_requires = '>=0.5.0'
+    name = "Qulacs device"
+    short_name = "qulacs.simulator"
+    pennylane_requires = ">=0.5.0"
     version = __version__
-    author = 'Steven Oud and Xanadu'
+    author = "Steven Oud and Xanadu"
+    gpu_supported = GPU_SUPPORTED
 
     _capabilities = {
-        'model': 'qubit',
-        'tensor_observables': True
+        "model": "qubit",
+        "tensor_observables": True,
+        "inverse_operations": True
     }
 
-    _operations_map = {
-        'QubitStateVector': None,
-        'BasisState': None,
-        'QubitUnitary': None,
-        'Toffoli': toffoli,
-        'CSWAP': CSWAP,
-        'CRZ': crz,
-        'Rot': None,
-        'SWAP': gate.SWAP,
-        'CNOT': gate.CNOT,
-        'CZ': gate.CZ,
-        'S': gate.S,
-        'Sdg': gate.Sdag,
-        'T': gate.T,
-        'Tdg': gate.Tdag,
-        'RX': gate.RX,
-        'RY': gate.RY,
-        'RZ': gate.RZ,
-        'PauliX': gate.X,
-        'PauliY': gate.Y,
-        'PauliZ': gate.Z,
-        'Hadamard': gate.H
+    _operation_map = {
+        "QubitStateVector": None,
+        "BasisState": None,
+        "QubitUnitary": None,
+        "Toffoli": gate.TOFFOLI,
+        "CSWAP": gate.FREDKIN,
```

Comment:
Added the two new gates that were just [released yesterday](https://github.com/qulacs/qulacs/releases/tag/v0.1.10). :tada: